src/main/java/eu/bbmri_eric/negotiator/common/AuthenticatedUserContext.java

```diff
@@ -15,9 +17,14 @@ public class AuthenticatedUserContext {
    *
    * @return the internal identifier
    */
-  public static Long getCurrentlyAuthenticatedUserInternalId() throws ClassCastException {
-    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-    return ((NegotiatorPrincipal) auth.getPrincipal()).getPerson().getId();
+  public static Long getCurrentlyAuthenticatedUserInternalId()
+      throws AuthenticationCredentialsNotFoundException {
+    try {
+      Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+      return ((NegotiatorPrincipal) auth.getPrincipal()).getPerson().getId();
+    } catch (Exception e) {
```

Comment:
please use the specific exception

Reply:
why? Don't we want to catch all exceptions? the result is the same, no authenticated user can be parsed?

Reply:
https://stackoverflow.com/questions/21938/is-it-really-that-bad-to-catch-a-general-exception

Reply:
added logging